### PR TITLE
Mobile queue fixes

### PR DESCRIPTION
**Pull request by aliabid94**
Closes: #2075

### CODE REVIEW

File: gradio/helpers.py [hunk 1]
```diff
@@ -893,3 +896,50 @@ def __init__(self, target: Block | None, _data: Any):
...
+    """
+    This class allows you to pass custom info messages to the user. You can do so simply with `gr.Info('message here')`, and when that line is executed the custom message will appear in a modal on the demo.
+    """
+
+    def __init__(self, message: str = "Info issued."):
+        """
+        Parameters:
+            message: The info message to be displayed to the user.
+        """
+        log_message(message, level="info")
```
Comment (akx) [inline gradio/helpers.py:945]: This API seems pretty weird – why are these classes? (Also, does it make sense for them to have a default message?)

File: gradio/queueing.py [hunk 2]
```diff
@@ -415,6 +453,8 @@ async def process_events(self, events: list[Event], batch: bool) -> None:
             end_time = time.time()
             if response.status == 200:
                 self.update_estimation(end_time - begin_time)
+        except Exception as e:
+            print(e)
```
Comment (akx) [inline gradio/queueing.py:457]: 😨

File: gradio/helpers.py (hunk 1 above)
Comment (aliabid94) [reply gradio/helpers.py]: I had set the wrong target branch - you probably want to comment here: https://github.com/gradio-app/gradio/pull/4518

**Comment by abidlabs** [comment]
Working on testing this

File: js/app/src/Blocks.svelte [hunk 3]
```diff
@@ -240,12 +240,30 @@
...
 	const MESSAGE_QUOTE_RE = /^'([^]+)'$/;
 
 	const DUPLICATE_MESSAGE =
 		"There is a long queue of requests pending. Duplicate this Space to skip.";
+	const MOBILE_QUEUE_WARNING =
+		"On mobile, the connection can break if this tab is unfocused or the device sleeps, losing your position in queue.";
+	const MOBILE_RECONNECT_MESSAGE =
+		"Lost connection due to leaving page. Rejoining queue...";
 	const SHOW_DUPLICATE_MESSAGE_ON_ETA = 15;
+	const SHOW_MOBILE_QUEUE_WARNING_ON_ETA = 10;
```
Comment (abidlabs) [inline js/app/src/Blocks.svelte:260]: Perhaps also set this to 15 seconds so that we have a consistent definition of a "long waiting time" that we can communicate to users

**Comment by abidlabs** [comment]
Very nice @aliabid94! Can confirm that I see this warning modal when I initiate the job:

image

(though only with queuing enabled, which I think is fine)

And I see this when I switch tabs. (Btw on my iPhone, I only see this if I set it to low power mode, otherwise the connection doesn't break):

image

**Review by abidlabs — APPROVED** [review]
LGTM!

**Comment by freddyaboulton** [comment]
Maybe this is part of the scroll_to_output problem on spaces but on the deployed demo the warning gets automatically closed 


https://github.com/gradio-app/gradio/assets/41651716/d74a87da-bc9f-4d04-8b72-c014b80d2114

**Comment by abidlabs** [comment]
Hmm @freddyaboulton it's working fine for me on iPhone (tested on Chrome and Safari)

image

Are you on iPhone?

**Comment by freddyaboulton** [comment]
yes - iphone and on safari 🤔 I have an ancient iphone 10 though. let me try again.

File: js/app/src/Blocks.svelte (hunk 3 above)
Comment (freddyaboulton) [reply js/app/src/Blocks.svelte:260]: Makes sense

**Review by freddyaboulton — COMMENTED** [review]
Modal stopped disappearing for me @aliabid94 ! Thanks for the fix

File: js/app/src/Blocks.svelte (hunk 3 above)
Comment (aliabid94) [reply js/app/src/Blocks.svelte:260]: I chose 10 seconds specifically because that's how long it seems iPhones break a websocket connection on low power mode. I don't think there's any need for these to match

Comment (akx) [reply js/app/src/Blocks.svelte:260]: That should probably be documented in a comment near the constant :)
